planner/core/rule_join_elimination.go

```diff
@@ -28,40 +29,44 @@ type outerJoinEliminator struct {
 // 2. outer join elimination with duplicate agnostic aggregate functions: For example left outer join.
 //    If the parent only use the columns from left table with 'distinct' label. The left outer join can
 //    be eliminated.
-func (o *outerJoinEliminator) tryToEliminateOuterJoin(p *LogicalJoin, aggCols []*expression.Column, parentSchema *expression.Schema) (LogicalPlan, error) {
+func (o *outerJoinEliminator) tryToEliminateOuterJoin(p *LogicalJoin, aggCols []*expression.Column, parentCols []*expression.Column) (LogicalPlan, bool, error) {
 	var innerChildIdx int
 	switch p.JoinType {
 	case LeftOuterJoin:
 		innerChildIdx = 1
 	case RightOuterJoin:
 		innerChildIdx = 0
 	default:
-		return p, nil
+		return p, false, nil
 	}
 
 	outerPlan := p.children[1^innerChildIdx]
 	innerPlan := p.children[innerChildIdx]
+	outerUniqueIDs := set.NewInt64Set()
+	for _, outerCol := range outerPlan.Schema().Columns {
+		outerUniqueIDs.Insert(outerCol.UniqueID)
+	}
+	matched := o.isColsAllFromOuterTable(outerPlan, parentCols, outerUniqueIDs)
+	if !matched {
+		return p, false, nil
+	}
 	// outer join elimination with duplicate agnostic aggregate functions
-	matched, err := o.isAggColsAllFromOuterTable(outerPlan, aggCols)
-	if err != nil || matched {
-		return outerPlan, err
+	matched = o.isColsAllFromOuterTable(outerPlan, aggCols, outerUniqueIDs)
+	if matched {
+		return outerPlan, true, nil
 	}
 	// outer join elimination without duplicate agnostic aggregate functions
-	matched, err = o.isParentColsAllFromOuterTable(outerPlan, parentSchema)
-	if err != nil || !matched {
-		return p, err
-	}
 	innerJoinKeys := o.extractInnerJoinKeys(p, innerChildIdx)
 	contain, err := o.isInnerJoinKeysContainUniqueKey(innerPlan, innerJoinKeys)
```

Comment:
Seems it is duplicated with `extractInnerJoinKeys()`, can we remove it?

Reply:
We can't. `innerJoinKeys` appears not only here.